ledger/eval/eval.go

```diff
@@ -1637,18 +1742,25 @@ func (eval *BlockEvaluator) generateKnockOfflineAccountsList() {
 					updates.ExpiredParticipationAccounts,
 					accountAddr,
 				)
-				continue // if marking expired, do not also suspend
+				continue // if marking expired, do not consider suspension
 			}
 		}
 
+		// Absent check: has it been too long since the last heartbeat/proposal, or
+		// has this online account failed a challenge?
 		if len(updates.AbsentParticipationAccounts) >= maxSuspensions {
 			continue // no more room (don't break the loop, since we may have more expiries)
 		}
 
-		if acctData.Status == basics.Online {
+		if acctData.Status == basics.Online && acctData.IncentiveEligible {
 			lastSeen := max(acctData.LastProposed, acctData.LastHeartbeat)
-			if isAbsent(eval.state.prevTotals.Online.Money, acctData.MicroAlgos, lastSeen, current) ||
-				failsChallenge(ch, accountAddr, lastSeen) {
+			oad, lErr := eval.state.lookupAgreement(accountAddr)
+			if lErr != nil {
+				logging.Base().Errorf("unable to check account for absenteeism: %v", accountAddr)
```

Comment:
This should never happen right? Not being able to check an account would let them live on eternally (think we discussed this exact case live a few weeks ago).